### ISSUE TITLE
Incorrect error inspection when returning 'void' in a union type?

### ISSUE THREAD

**Issue by pouwelsjochem** [Benjamin-Dobell/IntelliJ-Luanalysis#59]
Environment
--------------
   name   |   version  
----------|----------
 IDEA version         | IC2020.3
 Luanalysis version | 1.2.2
 OS                        | MacOS Big Sur

What are the steps to reproduce this issue?
------------------------------------------------
Create an `@param` with a fun():void|number
```
---@param _function1 fun()
---@param _function2 fun():void
---@param _function3 fun():any|number
---@param _function4 fun():any
---@param _function5 fun():void|number ERROR
---@param _function6 fun():void|number|nil ERROR
---@param _function7 (fun():number)|(fun():any)
---@param _function8 (fun():number)|(fun():void)
local function optionalReturnValueTest(_function1, _function2, _function3, _function4, _function5, _function6, _function7, _function8)

end
optionalReturnValueTest(function()

end, function()

end, function()

end, function()

end, function()

end, function()

end, function()

end, function()

end)
```

What happens?
-------------
An error is indicated at all methods where `void` is used to indicate an optional `number` return value. When using `any` it doesn't give an error anymore, but its a very confusing way to indicate a function with an optional `number` return value.

The `(fun():number)|(fun():void)` sounds like the best alternative to me, so people reading this issue might consider that as well.

What were you expecting to happen?
-----------------------------------------
No error when there's no return statement when using `void|number` as a return value.

**Comment by adriweb** [comment]
See #45

**Comment by Benjamin-Dobell** [comment]
Closing in favour of #45
